arrebol/worker/auth/token/jwt.go

```diff
@@ -40,41 +32,72 @@ func NewJWToken(worker *worker.Worker) (*JWToken, error){
 		},
 	}
 	token := jwt.NewWithClaims(jwt.SigningMethodRS512, claims)
-	privateKey, err := crypto.GetPrivateKey(os.Getenv(ArrebolPrivKeyPath))
+	signedToken, err := signToken(token)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
-	tokenStr, err := token.SignedString(privateKey)
+	return JWToken(signedToken), nil
+}
+
+func signToken(token *jwt.Token) (string, error) {
+	privateKey, err := crypto.GetPrivateKey(os.Getenv(ArrebolPrivKeyPath))
 	if err != nil {
-		return &JWToken{}, err
+		return "", err
 	}
-	return &JWToken{
-		raw:      tokenStr,
-		QueueId:  worker.QueueId,
-		WorkerId: worker.ID,
-	}, nil
+	return token.SignedString(privateKey)
 }
 
-func Parse(tokenString string) (*JWToken, error) {
+func Parse(tokenString string) (*jwt.Token, error) {
 	token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
 		return crypto.GetPublicKey(os.Getenv(ArrebolPubKeyPath))
 	})
 	if err != nil {
 		return nil, err
 	}
-	log.Println(token)
-	return nil, nil
+	return token, nil
 }
 
-func (t *JWToken) String() string {
-	return t.raw
+func (t JWToken) String() string {
+	return string(t)
 }
 
-func (*JWToken) Expired() bool {
+func (t JWToken) Expired() bool {
+	_, err := Parse(t.String())
+	v, _ := err.(*jwt.ValidationError)
+
+	if v.Errors == jwt.ValidationErrorExpired {
+		return true
+	}
+	if err != nil {
+		panic(err)
+	}
 	return false
 }
 
-func (*JWToken) GetPayload(key string) string {
-	return ""
+func (t JWToken) GetPayload(key string) (interface{}, error) {
```

Comment:
This name gives the idea that the whole payload is going to be returned, but only one field of it is returned. Maybe you should rename it to GetPayloadField or something better. Same thing for SetPayload

Reply:
Done.